src/sage/algebras/hecke_algebras/ariki_koike_specht_modules.py

```diff
@@ -0,0 +1,473 @@
+            sage: S._L_on_basis(9, t)
+            (u2*q^-2)*S[([[2, 4], [8]], [], [[1, 3, 7], [5, 6], [9, 10]])]
+        """
+        c = t.cells_containing(i)[0]
+        if len(c) == 2: # it is of level 1 and a regular tableau
+            c = (0,) + c
+        res = self._q**(c[2] - c[1]) * self._u[c[0]]
+        R = self.base_ring()
+        return self.element_class(self, {t: R(res)})
+
+    def _T_on_basis(self, i, t):
+        r"""
+        Return the action of `T_i` on the basis element indexed by
+        the standard tableau tuple ``t``.
+
+        EXAMPLES::
+
+            sage: AK = algebras.ArikiKoike(3, 10, use_fraction_field=True)
+            sage: S = AK.specht_module([[2,1], [], [3,2,2]])
+            sage: P = S.basis().keys()
+            sage: t = P([[[2,4],[8]], [], [[1,5,7],[3,6],[9,10]]])
+            sage: S._T_on_basis(0, t) == S._L_on_basis(1, t)
+            True
+            sage: S._T_on_basis(1, t)
+            ((u2*q-u0)/(u0-u2))*S[([[1, 4], [8]], [], [[2, 5, 7], [3, 6], [9, 10]])]
+             + ((u0*q-u0)/(u0-u2))*S[([[2, 4], [8]], [], [[1, 5, 7], [3, 6], [9, 10]])]
+            sage: S._T_on_basis(2, t)
+            ((u2*q-u2)/(-u0*q+u2))*S[([[2, 4], [8]], [], [[1, 5, 7], [3, 6], [9, 10]])]
+             + ((u0*q^2-u2)/(-u0*q+u2))*S[([[3, 4], [8]], [], [[1, 5, 7], [2, 6], [9, 10]])]
+            sage: S._T_on_basis(5, t)
+            -S[([[2, 4], [8]], [], [[1, 5, 7], [3, 6], [9, 10]])]
+            sage: S._T_on_basis(7, t)
+            ((u2*q^4-u0)/(-u2*q^3+u0))*S[([[2, 4], [7]], [], [[1, 5, 8], [3, 6], [9, 10]])]
+             + ((u0*q-u0)/(-u2*q^3+u0))*S[([[2, 4], [8]], [], [[1, 5, 7], [3, 6], [9, 10]])]
+            sage: S._T_on_basis(9, t)
+            q*S[([[2, 4], [8]], [], [[1, 5, 7], [3, 6], [9, 10]])]
+        """
+        R = self.base_ring()
+        if i == 0:
+            return self._L_on_basis(1, t)
+
+        ct = t.cells_containing(i)[0]
+        cs = t.cells_containing(i+1)[0]
+        if len(ct) == 2: # it is of level 1 and a regular tableau
+            ct = (0,) + ct
+            cs = (0,) + cs
+
+        if ct[0] == cs[0] and ct[2] == cs[2]: # same column
+            return self.element_class(self, {t: -R.one()})
+
+        if ct[0] == cs[0] and ct[1] == cs[1]: # same row
+            return self.element_class(self, {t: self._q})
+
+        # result is standard
+        s = t.symmetric_group_action_on_entries(self._Pn.simple_reflection(i))
+        assert s.parent() is t.parent()
+
+        def res(cell):
+            return self._q**(cell[2] - cell[1]) * self._u[cell[0]]
+
+        # Note that the residue of i in t is given by the cell c
+        #   and of i in s corresponds to cell cp because the
+        #   corresponding action of the permutation on t.
+        one = self.base_ring().one()
+        denom = res(cs) - res(ct)
+        coefft = (self._q - one) * res(cs) / denom
+        coeffs = (self._q * res(ct) - res(cs)) / denom
```

Comment:
```suggestion
        coeffs = (self._q*res(ct) - res(cs)) / denom
```

Reply:
I find the spacing here is more readable considering the line above.

Reply:
ok